DataDog/libddprof
# Enhance license information

# What does this PR do?

- Add LICENSE-3rdparty.yml.
- Add LICENSE, LICENSE-3rdparty.yml, and NOTICE to the output of ffi-build.sh.
- Commits Cargo.lock.

# Motivation

Datadog likes all open-source repositories to track and present 3rd-party licenses. This is the last remaining item before we can go public with this repository. To keep version numbers stable, we commit the Cargo.lock.

# Additional Notes

This LICENSE-3rdparty.yml file is generated with this command:

```
cargo bundle-licenses --format yaml --output LICENSE-3rdparty.yml
```

And it can be checked in CI output like so:

```
cargo bundle-licenses \
    --format yaml \
    --output /tmp/CI.yaml \
    --previous LICENSE-3rdparty.yml \
    --check-previous
```

# How to test the change?

Check the downstream jobs in the libddprof-build repository on branch `levi/cargo-bundle-license`.
